aoc-2022-rust/src/days/day_13/main_13.rs

```diff
@@ -25,20 +28,26 @@ fn part_2(input: &str) -> usize {
     list.push(SignalValue::Array(vec![SignalValue::Array(vec![
         SignalValue::Number(6),
     ])]));
-    list.sort_by(|left, right| compare_signal_pair(left, right));
+    list.sort_by(compare_signal_pair);
     let mut index_2: Option<usize> = None;
     let mut index_6: Option<usize> = None;
     for (i, ele) in list.into_iter().enumerate() {
         if !ele.is_num_value() {
             let list_1 = ele.borrow_list().unwrap();
+
```

Comment:
This entire for loop is pretty messy.
I rewrote it using the position function you mentioned earlier.